common/cluster/metadata.go

```diff
@@ -345,15 +343,15 @@ func (m *metadataImpl) GetAllClusterInfo() map[string]ClusterInformation {
 	return result
 }
 
-func (m *metadataImpl) ClusterNameForFailoverVersion(isGlobalNamespace bool, failoverVersion int64) (string, error) {
+func (m *metadataImpl) ClusterNameForFailoverVersion(isGlobalNamespace bool, failoverVersion int64) string {
 	if failoverVersion == common.EmptyVersion {
 		// Local namespace uses EmptyVersion. But local namespace could be promoted to global namespace. Once promoted,
 		// workflows with EmptyVersion could be replicated to other clusters. The receiving cluster needs to know that
 		// those workflows are not from their current cluster.
 		if isGlobalNamespace {
-			return FakeClusterForEmptyVersion, nil
+			return unknownClusterNamePrefix + strconv.Itoa(int(failoverVersion))
```

Comment:
this needs to be initial failover version

Reply:
failoverVersion == common.EmptyVersion

Reply:
oh, sorry.